### PR TITLE
Make UWP clrcompression package aot specific

### PR DESCRIPTION
**Pull request by ianhays**
This commit modifies the packaging of the ClrCompression native windows binary to work for all RIDs and frameworks. Previously, using the win10 package on a non-UWP program would fail becase there was only one native asset in the package: the App-container one. Becase of [NuGet limitations](https://github.com/NuGet/Home/issues/2782) we couldn't put regular binaries alongside the AppX one, so we distinguished AppX vs non-AppX by the win10 vs win7 RID. With this commit we'll only ship the AOT package with the APPX clrcompression that is built internally; all other RIDS and Frameworks will get the same clrcompression version that is built from the Open.

This change should *just work* with the packaging infrastructure in TFS as well, since @jhendrixMSFT recently changed it to build the win10 package which is altered in this commit to produce AOT specific packages.

resolves https://github.com/dotnet/corefx/issues/7436

@ericstj

### CODE REVIEW

File: src/Native/pkg/runtime.native.System.IO.Compression/win10/runtime.native.System.IO.Compression.pkgproj [hunk 1]
```diff
@@ -4,12 +4,12 @@
...
   <PropertyGroup>
     <Version>4.0.1</Version>
-    <PackageTargetRuntime>win10-$(PackagePlatform)</PackageTargetRuntime>
+    <PackageTargetRuntime>win10-$(PackagePlatform)-aot</PackageTargetRuntime>
   </PropertyGroup>
 
   <ItemGroup>
     <NativeFile Include="$(WinNativePath)..\..\NetCoreForCoreCLR\native\clrcompression.dll">
-      <TargetPath>runtimes/win10-$(PackagePlatform)/lib/netcore50</TargetPath>
+      <TargetPath>runtimes/win10-$(PackagePlatform)-aot/lib/netcore50</TargetPath>
```
Comment (ericstj) [inline src/Native/pkg/runtime.native.System.IO.Compression/win10/runtime.native.System.IO.Compression.pkgproj]: We could change this back to native now, but I'm ok leaving as is.  Either way this package will need an update once CoreRT starts using the *aot* monikers.  CoreRT will be outside of AppX.  I expect by that time we'll have the native+tfm feature in NuGet and we can refactor to use that.

**Comment by ericstj** [comment]
LGTM.  Thanks for getting this in!

**Comment by ianhays** [comment]
Thanks Eric! Shiproom request coming in a few.

**Comment by weshaggard** [comment]
@ericstj @ianhays just so I'm clear with this change now resolving for win10-x64 will pick-up the win7-x64 asset? It really isn't very clear to me why that wasn't the case before given that win10 asset was in a lib\netcore50 only folder.

**Comment by ericstj** [comment]
Before we had a win10 specific package with no TFM agnostic clrcompression.  Now we have win10-aot package so win10 non-aot won't get this and will fallback to the win7/8 specific packages.

**Comment by GuardRex** [comment]
> This commit modifies the packaging of the ClrCompression native windows binary to work for all RIDs and frameworks

@ericstj With this merge, we're clear to use `win10-x64` for Win10/WS2016/Nano, correct?

**Comment by ericstj** [comment]
Yeah it should work now.

**Pull request merged**